include/pstore/exchange/import_compilation.hpp

```diff
@@ -62,254 +62,247 @@
 
 namespace pstore {
     namespace exchange {
+        namespace import {
+
+            using fragment_index_pointer = std::shared_ptr<index::fragment_index>;
+
+            //*     _      __ _      _ _   _           *
+            //*  __| |___ / _(_)_ _ (_) |_(_)___ _ _   *
+            //* / _` / -_)  _| | ' \| |  _| / _ \ ' \  *
+            //* \__,_\___|_| |_|_||_|_|\__|_\___/_||_| *
+            //*                                        *
+            //-MARK: definition
+            class definition final : public rule {
+            public:
+                using container = std::vector<repo::compilation_member>;
+
+                definition (not_null<context *> ctxt, not_null<container *> definitions,
+                            not_null<name_mapping const *> names,
+                            fragment_index_pointer const & fragments);
+                definition (definition const &) = delete;
+                definition (definition &&) noexcept = delete;
+
+                ~definition () noexcept override = default;
+
+                definition & operator= (definition const &) = delete;
+                definition & operator= (definition &&) noexcept = delete;
+
+                gsl::czstring name () const noexcept override { return "definition"; }
+
+                std::error_code key (std::string const & k) override;
+                std::error_code end_object () override;
+
+                static maybe<repo::linkage> decode_linkage (std::string const & linkage);
+                static maybe<repo::visibility> decode_visibility (std::string const & visibility);
+
+            private:
+                not_null<container *> const definitions_;
+                not_null<name_mapping const *> const names_;
+                fragment_index_pointer const fragments_;
+
+                enum { digest_index, name_index, linkage_index, visibility_index };
+                std::bitset<visibility_index + 1> seen_;
+
+                std::string digest_;
+                std::uint64_t name_ = 0;
+                std::string linkage_;
+                std::string visibility_;
+            };
+
+
+            //*     _      __ _      _ _   _                _     _        _    *
+            //*  __| |___ / _(_)_ _ (_) |_(_)___ _ _    ___| |__ (_)___ __| |_  *
+            //* / _` / -_)  _| | ' \| |  _| / _ \ ' \  / _ \ '_ \| / -_) _|  _| *
+            //* \__,_\___|_| |_|_||_|_|\__|_\___/_||_| \___/_.__// \___\__|\__| *
+            //*                                                |__/             *
+            //-MARK: definition object
+            class definition_object final : public rule {
+            public:
+                definition_object (not_null<context *> ctxt,
+                                   not_null<definition::container *> definitions,
+                                   not_null<name_mapping const *> names,
+                                   fragment_index_pointer const & fragments);
```

Comment:
Missing `const` for parameters?

definition_object (not_null<context *> **const** ctxt,
                                   not_null<definition::container *> **const** definitions,
                                   not_null<name_mapping const *> **const** names,
                                   fragment_index_pointer const & fragments);

Reply:
No (as above).